src/Sarif/Readers/IRuleConverter.cs

```diff
@@ -44,11 +44,15 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
                     Id = iRule.Id,
                     MessageFormats = iRule.MessageFormats,
                     Name = iRule.Name,
-                    Properties = iRule.Properties,
                     ShortDescription = iRule.ShortDescription,
                     Tags = iRule.Tags
                 };
 
+                foreach (string propertyName in iRule.PropertyNames)
+                {
+                    rule.SetProperty(propertyName, iRule.GetProperty(propertyName));
+                }
```

Comment:
We might consider a adding a convenience method `IPropertyBagHolder.CopyPropertiesFrom(IPropertyBagHolder other)`.